File: src/main/kotlin/baseprovidedbyuni/bookquery/JsonResult.java
package baseprovidedbyuni.bookquery;

import com.google.gson.annotations.SerializedName;
import java.util.List;

/**
 * Created by Arno on 23-3-2017.
 */
public class JsonResult {

    @SerializedName("kind")
    private String kind;
    @SerializedName("totalItems")
    private int totalItems;
    @SerializedName("items")
    private List<Book> books;

    public Book getBook() {
        return books.get(0);
    }

    public BookDetail getBookDetail() {
        return books.get(0).getBookDetail();
    }

}

File: src/main/kotlin/com/pkg_dot_zip/bookquery/JsonResult.kt
package com.pkg_dot_zip.com.pkg_dot_zip.bookquery

import com.google.gson.annotations.SerializedName

class JsonResult {
    @SerializedName("kind")
    private val kind: String? = null

    @SerializedName("totalItems")
    private val totalItems = 0

    @SerializedName("items")
    private val books: List<Book>? = null

    fun getBook(): Book? = books?.first()

    fun getBookDetail(): BookDetail = books?.first()?.bookDetail!!
}